kong-ngx-build

```diff
@@ -154,8 +154,8 @@ main() {
   OPENSSL_PREFIX=${OPENSSL_PREFIX:-$PREFIX/openssl}
   OPENRESTY_PREFIX=${OPENRESTY_PREFIX:-$PREFIX/openresty}
 
-  OPENSSL_INSTALL=$(realpath -m $OPENSSL_DESTDIR/$OPENSSL_PREFIX)
-  OPENRESTY_INSTALL=$(realpath -m $OPENRESTY_DESTDIR/$OPENRESTY_PREFIX)
+  OPENSSL_INSTALL=$(readlink -f $OPENSSL_DESTDIR/$OPENSSL_PREFIX)
```

Comment:
By the reading of man docs, seems to me `readlink -e` might be better.

Reply:
I'm not so sure.

```
-e, --canonicalize-existing
canonicalize by following every symlink in every component of the given name recursively, all components must exist

-m, --canonicalize-missing
canonicalize by following every symlink in every component of the given name recursively, without requirements on components existence
```

I think we want the one where if the destination directory doesn't exist it works? I'll try and I might include a test to make sure we don't allow `//` to break things again

Reply:
Confirmed we want / need `-m`

```
readlink -e ///build/openssl
echo $?
1
```

Reply:
Seems the `readlink` on centos6 doesn't behave this way though

```
root@a00fa19261e8:/src/openresty-build-tools# readlink -f /tmp/build//tmp/build/usr/local/openssl
root@a00fa19261e8:/src/openresty-build-tools# echo $?
1
root@a00fa19261e8:/src/openresty-build-tools# readlink -e /tmp/build//tmp/build/usr/local/openssl
root@a00fa19261e8:/src/openresty-build-tools# echo $?
1
```

Reply:
So on centos6 the directory needs to exist ahead of time and I can't find any important difference between `-f / -e`